persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideActor.scala

```diff
@@ -122,4 +161,34 @@ private[lagom] class ReadSideActor[Event <: AggregateEvent[Event]](
       // This actor will be restarted by WorkerCoordinator
       throw cause
   }
+
+  private def userFlowWrapper(
+      workerCoordinates: WorkerCoordinates,
+      userFlow: Flow[japi.Pair[Event, LagomOffset], Done, _]
+  ): Flow[(japi.Pair[Event, LagomOffset], AkkaOffset), AkkaOffset, _] =
+    Flow.fromGraph(GraphDSL.create(userFlow) { implicit builder => wrappedFlow =>
+      import GraphDSL.Implicits._
+      val unzip = builder.add(Unzip[japi.Pair[Event, LagomOffset], AkkaOffset])
+      val zip   = builder.add(Zip[Done, AkkaOffset])
+      val metricsReporter: FlowShape[(Done, AkkaOffset), AkkaOffset] = builder.add(Flow.fromFunction {
+        e: (Done, AkkaOffset) =>
```

Comment:
Here we can make it nicer by using case syntax with tuple

```scala
case (_, offset) => ... // use are not using the Done anyway
```

Then later instead of `e._2`, we use `offset`.